esphome/components/bl0940/button/calibration_reset_button.cpp

```diff
@@ -0,0 +1,22 @@
+#include "calibration_reset_button.h"
+#include "../bl0940.h"
+#include "esphome/core/hal.h"
+#include "esphome/core/log.h"
+#include "esphome/core/application.h"
+
+namespace esphome {
+namespace bl0940 {
+
+static const char *const TAG = "calibration_reset.button";
+
+void CalibrationResetButton::dump_config() { LOG_BUTTON("", "Calibration Reset Button", this); }
+
+void CalibrationResetButton::press_action() {
+  ESP_LOGI(TAG, "Resetting calibration defaults...");
+  if (bl0940_ != nullptr) {
+    bl0940_->reset_calibration();
+  }
```

Comment:
With the above change to parent, plus not possible to be null due to validations
```suggestion
  this->parent_->reset_calibration();
```